### ISSUE TITLE
CVE-2023-23397.ps1 fails if AzureADPreview is installed instead of AzureAD module

### ISSUE THREAD

**Issue by hbrowne22** [microsoft/CSS-Exchange#1567]
.\CVE-2023-23397.ps1 -CreateAzureApplication fails with an error:

```
CVE-2023-23397 script version 23.03.14.1625
Unable to connect to Azure AD... Make sure you have AzureAD module installed Inner Exception

AzureAD module was not found on this computer. You can install it by running Install-Module AzureAD
```

if the module AzureADPreview is installed instead of AzureAD.

The script checks for the presence of the AzureAD module in 3 functions: I replaced these references with AzureADPreview:

```
if ($null -ne (Get-InstalledModule -Name "AzureAD" -ErrorAction SilentlyContinue)) {
                Import-Module "AzureAD" -ErrorAction Stop
```
to:

```
if ($null -ne (Get-InstalledModule -Name "AzureADPreview" -ErrorAction SilentlyContinue)) {
                Import-Module "AzureADPreview" -ErrorAction Stop
```

Although obviously it would be neater to check for both modules!

**Comment by bill-long** [comment]
The plan is to remove the AzureAD dependency entirely, so this will be fixed by #1562 if not before.

**Comment by lusassl-msft** [comment]
As discussed with engineering, we will not support the Azure AD preview module. Plan is to get rid of the Azure AD module in general as it will be deprecated this year and is also not supported to work with PowerShell Core.
